src/metrics-parse.go

```diff
@@ -126,3 +151,53 @@ func populatePartialMetrics(ms *metric.Set, metrics map[string]interface{}, metr
 		}
 	}
 }
+
+func collectVersion(db dataSource) (*semver.Version, error) {
+	versionQueryResult, err := db.query(versionQuery)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching version: %w", err)
+	}
+
+	if versionStr, exists := versionQueryResult["version"]; exists {
+		version := versionStr.(string)
+		log.Debug("Original MySQL Server version string: %s", version)
+
+		sanitizedVersion, err := extractSemver(version)
+		if err != nil {
+			return nil, fmt.Errorf("error extracting version: %w", err)
+		}
+
+		semVersion, err := semver.ParseTolerant(sanitizedVersion)
+		if err != nil {
+			return nil, err
+		}
+
+		log.Debug("Parsed version as semver: %v", semVersion)
+		return &semVersion, nil
+	} else {
+		return nil, fmt.Errorf("%w", errVersionNotFound)
+	}
+}
+
+// extractSemver uses a regular expression to extract a semver-compliant version string up to major.minor.patch
+func extractSemver(version string) (string, error) {
+	reg := regexp.MustCompile(`^(?P<major>\d+)(?:\.(?P<minor>\d+))?(?:\.(?P<patch>\d+))?`)
+	matches := reg.FindStringSubmatch(version)
+
+	if len(matches) == 0 || matches[1] == "" {
+		return "", errSemanticVersionNotFound
+	}
+
+	major := matches[1]
+	minor := "0"
+	patch := "0"
+
+	if len(matches) > 2 && matches[2] != "" {
+		minor = matches[2]
+	}
+	if len(matches) > 3 && matches[3] != "" {
+		patch = matches[3]
+	}
+
+	return fmt.Sprintf("%s.%s.%s", major, minor, patch), nil
+}
```

Comment:
I don't think we need this much complexity for getting the version. [MySQL Doc](https://dev.mysql.com/doc/refman/8.4/en/information-functions.html#function_version) specifies that the version might have a suffix but not a prefix. Therefore, you'll always get version before the first `-`.
Moreover, the documentation states: `This function is unsafe for statement-based replication. A warning is logged if you use this function when [binlog_format](https://dev.mysql.com/doc/refman/8.4/en/replication-options-binary-log.html#sysvar_binlog_format) is set to STATEMENT.` Though the `binlog_format` is deprecated, do you think it might cause any issues for older customer environments?

Reply:
will discuss this with kai

Reply:
I ran integration tests in github actions for version 5.7.35 with binlog_format=STATEMENT and it [passed](https://github.com/newrelic/nri-mysql/actions/runs/12559212682/job/35014599864?pr=170). I didn’t make any change to handle if we are not able to parse DB version output. So, We are getting the version as excepted without any warning with was mentioned in the Mysql doc.